packages/api/apollo-server/mutations/assignment.ts

```diff
@@ -0,0 +1,40 @@
+import { assignments } from '../mock-data'
+
+export interface IAddAssignment {
+  id: String
+  name: String
+  requiredFiles?: String[]
+  peerReviewDeadline?: String
+  feedbackQuestions?: String[]
+  groupSize?: Number
+  isActive?: Boolean
+}
+
+// add a new assignment
+export const addAssignment = (_: any, args: IAddAssignment) => {
+  console.log(args)
+  return null
+}
+
+export interface IEditAssignment {
+  id: String
+  name?: String
+  requiredFiles?: String[]
+  peerReviewDeadline?: String
+  feedbackQuestions?: String[]
+  groupSize?: Number
+  isActive?: Boolean
+}
```

Comment:
IEditSubmission should follow IAddSubmission but with an extra id field. If X is required in creation, then it should be required in editing. We will just do a simple replacement when updating the db.